cmd/account/generate-secret.go

```diff
@@ -0,0 +1,204 @@
+		return cmdutil.UsageErrorf(cmd, "AWS account CR name and AWS account ID cannot be empty at the same time")
+	}
+
+	if o.accountName != "" && o.accountID != "" {
+		return cmdutil.UsageErrorf(cmd, "AWS account CR name and AWS account ID cannot be set at the same time")
+	}
+
+	// only initialize kubernetes client when account name is set
+	if o.accountName != "" {
+		var err error
+		o.kubeCli, err = k8s.NewClient(o.flags)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *generateSecretOptions) run() error {
+	ctx := context.TODO()
+	var err error
+	awsSetupClient, err := awsprovider.NewAwsClient(o.profile, o.region, o.cfgFile)
+	if err != nil {
+		return err
+	}
+
+	var accountID string
+	if o.accountName != "" {
+		account, err := k8s.GetAWSAccount(ctx, o.kubeCli, o.accountNamespace, o.accountName)
+		if err != nil {
+			return err
+		}
+		accountID = account.Spec.AwsAccountID
+	} else {
+		accountID = o.accountID
+	}
+
+	callerIdentityOutput, err := awsSetupClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err != nil {
+		return err
+	}
+
+	roleArn := aws.String(fmt.Sprintf("arn:aws:iam::%s:role/%s", accountID, awsv1alpha1.AccountOperatorIAMRole))
+	credentials, err := awsprovider.GetAssumeRoleCredentials(awsSetupClient, aws.Int64(900),
+		callerIdentityOutput.UserId, roleArn)
+	if err != nil {
+		return err
+	}
+
+	awsClient, err := awsprovider.NewAwsClientWithInput(&awsprovider.AwsClientInput{
+		AccessKeyID:     *credentials.AccessKeyId,
+		SecretAccessKey: *credentials.SecretAccessKey,
+		SessionToken:    *credentials.SessionToken,
+		Region:          o.region,
+	})
+	if err != nil {
+		return err
+	}
+
+	// if the account ID is specified, will
+	if o.accountID != "" {
+		callerIdentityOutput, err = awsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+		if err != nil {
+			return err
+		}
+		if *callerIdentityOutput.Account != o.accountID {
+			return fmt.Errorf("account ID %s does not match the input account ID %s",
+				*callerIdentityOutput.Account, o.accountID)
+		}
+	}
+
+	username := aws.String(o.iamUsername)
+	ok, err := awsprovider.CheckIAMUserExists(awsClient, username)
+	if err != nil {
+		return err
+	}
+
+	// the specified user not exist, create one
+	if !ok {
+		policyArn := aws.String("arn:aws:iam::aws:policy/AdministratorAccess")
+		if err := awsprovider.CreateIAMUserAndAttachPolicy(awsClient,
+			username, policyArn); err != nil {
+			return err
+		}
+	} else {
+		fmt.Fprintf(o.IOStreams.Out, "User %s exists, deleting existing access keys now.\n", o.iamUsername)
+		if err := awsprovider.DeleteUserAccessKeys(awsClient, username); err != nil {
+			return err
+		}
+	}
+
+	newKey, err := awsClient.CreateAccessKey(&iam.CreateAccessKeyInput{
+		UserName: username,
+	})
+	if err != nil {
+		return err
+	}
+
+	secret := k8s.NewAWSSecret(
+		o.secretName,
+		o.secretNamespace,
+		*newKey.AccessKey.AccessKeyId,
+		*newKey.AccessKey.SecretAccessKey,
+	)
+	if !o.printSecret {
```

Comment:
The name of that argument variable conveys the opposite behaviour, can we rename it to `o.quiet` or something equivalent?